idb/postgres/internal/encoding/encoding_test.go

```diff
@@ -84,3 +84,52 @@ func TestJSONEncoding(t *testing.T) {
 
 	assert.Equal(t, x, xx)
 }
+
+func TestSanitizeNull(t *testing.T) {
```

Comment:
I think the test I want to see is a round trip desanitize(sanitize(x)) == x; 
and that will fail for x := "foo\\u0000bar" which already has the escaped-zero code in it which will become actual zero on its way back out through desanitize.